src/main/webapp/app/pages/HomePage.tsx

```diff
@@ -36,46 +44,43 @@ export type ExtendedTypeaheadSearchResp = TypeaheadSearchResp & {
 @observer
 class HomePage extends React.Component<IHomeProps> {
   @observable keyword = '';
+  @observable levelTypeSelected = LEVEL_TYPES.TX;
+  private LevelGadgets = this.generateLevelGadgets();
 
-  private levelGadgets: {
-    title?: string;
-    description: string;
-    level: string;
-    linkoutLevel: string;
-    combinedLevels: string[];
-  }[] = [
-    {
-      level: '1',
-      description: LEVEL_BUTTON_DESCRIPTION['1'],
-      linkoutLevel: '1',
-      combinedLevels: ['1'],
-    },
-    {
-      level: '2',
-      description: LEVEL_BUTTON_DESCRIPTION['2'],
-      linkoutLevel: '2',
-      combinedLevels: ['2'],
-    },
-    {
-      level: '3',
-      description: LEVEL_BUTTON_DESCRIPTION['3'],
-      linkoutLevel: '3',
-      combinedLevels: ['3'],
-    },
-    {
-      level: '4',
-      description: LEVEL_BUTTON_DESCRIPTION['4'],
-      linkoutLevel: '4',
-      combinedLevels: ['4'],
-    },
-    {
-      level: 'R1',
-      description: 'Resistance',
-      title: 'Level R1/R2',
-      linkoutLevel: 'R1,R2',
-      combinedLevels: ['R1', 'R2'],
-    },
-  ];
+  generateLevelGadgets() {
+    const levelGadgets: {
+      title?: string;
+      description: string;
+      level: string;
+      linkoutLevel: string;
+      combinedLevels: string[];
+    }[] = [];
+    for (const ele in LEVELS) {
```

Comment:
Let's don't shorten the variable name when it's pretty short. Also `ele` does not make sense.